cmd/generate/cmd_util_test.go

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package main
+
+import (
+	"testing"
+
+	"github.com/nginxinc/nginx-go-crossplane/internal/generator"
+	"github.com/stretchr/testify/require"
+)
+
+//nolint:funlen
+func TestOverrideParser(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		input    string
+		expected overrideItem
+		wantErr  bool
+	}{
+		{
```

Comment:
nit: there's another idiom for table-driven tests that uses a `map` to specify the name of the test case, I think that reads a little better:

```golang
	testcases := map[string]struct{
		input    string
		expected overrideItem
		wantErr  bool
	}{
		"normalFormat_pass": {
			input: "location:ngxHTTPMainConf|ngxConfTake12,ngxStreamMainConf",
			expected: overrideItem{
				directive: "location",
				masks: []generator.Mask{
					{"ngxHTTPMainConf", "ngxConfTake12"},
					{"ngxStreamMainConf"},
				},
			},
		},
		// <other test cases omitted for brevity>
	}
	for name, tc := range testcases{
		tc := tc
		t.Run(name, func(t *testing.T) {
			t.Parallel()
			
			// <test omitted for brevity>
		})
	}}
```

Reply:
Oh yeah, this one is better and align with other ut in crossplane. Applied.

Reply:
@ryepup Many crossplane tests use a slice, but I agree that a `map` is better. You start to see hints of this when it becomes necessary to add a "name" or other key to the testcase struct.

@dengsh12 A more important benefit to using a `map` vs slice is that Go randomizes the order of keys when ranging over the `map`. I believe-- I do not have documentation that states this as fact on hand at the moment-- this is to dissuade anyone from getting used to maps being ordered in the event the internal details are changed. You can generally use this to your benefit as it will randomize the order the tests are run which can help to detect any unintentional shared memory that leaks through tests in the form of test flake.